File: C# Fundamentals/C# OOP Basics/11. InterfaceNAbstr/Ex/P07_FoodSh/Models/Person.cs
using System;

public class Person
{
    private string name;
    private int age;
    private string id;
    private DateTime birthdate;

    private string group;

    private int food;
    
    public string Group
    {
        get { return this.group; }
        private set { this.group = value; }
    }
    public string Id
    {
        get { return this.id; }
        private set { this.id = value; }
    }

    public string Name
    {
        get { return this.name; }
        private set { this.name = value; }
    }
    public int Age
    {
        get
        {
            return this.age;
        }
        private set
        {
            this.age = value;
        }
    }
    public DateTime Birthdate
    {
        get
        {
            return this.birthdate;
        }
        private set
        {
            this.birthdate = value;
        }
    }

    public int Food {
        get
        {
            return this.food;
        }
        set
        {
            this.food = value;
        }
    }

    public Person(string name, int age, string group)
    {
        this.Name = name;
        this.Age = age;
        this.Group = group;
    }

    public Person(string name, int age, string id, DateTime birthdate)
    {
        this.Name = name;
        this.Age = age;
        this.Id = id;
        this.Birthdate = birthdate;
    }
}

File: Java/01.Advanced/01.Advanced/17.IteratorsComparators/Ex/src/t_07_equality_logic/Person.java
package t_07_equality_logic;

import java.util.Objects;

public class Person implements Comparable<Person> {
    private final String name;
    private final int age;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Person person = (Person) o;
        return age == person.age &&
                name.equals(person.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, age);
    }

    @Override
    public int compareTo(Person o) {
        int comparison = this.name.compareTo(o.name);

        if (comparison == 0) {
            return Integer.compare(this.age, o.age);
        }

        return comparison;
    }
}
